File: app/src/main/java/com/zhufuc/pctope/Utils/CompressImage.java
package com.zhufuc.pctope.Utils;

import android.graphics.Bitmap;
import android.graphics.Matrix;

/**
 * Created by zhufu on 17-8-11.
 */

public class CompressImage {
    public static Bitmap getBitmap(Bitmap bitmap,int Height,int Width){
        Matrix matrix = new Matrix();
        int oldHeight = bitmap.getHeight(),oldWidth = bitmap.getWidth();
        if (oldHeight<=0 || oldWidth<=0)
            return null;

        Float scaleHeight = (float) Height/oldHeight,scaleWidth = (float) Width/oldWidth;

        matrix.postScale(scaleWidth,scaleHeight);

        return Bitmap.createBitmap(bitmap,0,0,oldWidth,oldHeight,matrix,false);
    }

    public static Bitmap getBitmap(Bitmap bitmap,float scale){
        Matrix matrix = new Matrix();
        int height = bitmap.getHeight(),width = bitmap.getWidth();
        matrix.postScale(scale,scale);
        return Bitmap.createBitmap(bitmap,0,0,width,height,matrix,false);
    }

    public static boolean testBitmap(int testWidth,int testHeight,Bitmap bitmap){
        return (bitmap.getHeight()>testHeight || bitmap.getWidth()>testWidth);
    }
}

File: app/src/main/java/com/zhufucdev/pctope/utils/CompressImage.kt
package com.zhufucdev.pctope.utils

import android.graphics.Bitmap
import android.graphics.Matrix

/**
 * Created by zhufu on 17-8-11.
 */

object CompressImage {
    fun getBitmap(bitmap: Bitmap, Height: Int, Width: Int): Bitmap? {
        val matrix = Matrix()
        val oldHeight = bitmap.height
        val oldWidth = bitmap.width
        if (oldHeight <= 0 || oldWidth <= 0)
            return null

        val scaleHeight = Height.toFloat() / oldHeight
        val scaleWidth = Width.toFloat() / oldWidth

        matrix.postScale(scaleWidth, scaleHeight)

        return Bitmap.createBitmap(bitmap, 0, 0, oldWidth, oldHeight, matrix, false)
    }

    fun getBitmap(bitmap: Bitmap, scale: Float): Bitmap {
        val matrix = Matrix()
        val height = bitmap.height
        val width = bitmap.width
        matrix.postScale(scale, scale)
        return Bitmap.createBitmap(bitmap, 0, 0, width, height, matrix, false)
    }

    fun testBitmap(testWidth: Int, testHeight: Int, bitmap: Bitmap): Boolean {
        return bitmap.height > testHeight || bitmap.width > testWidth
    }
}
